File: Medium/0789-Escape The Ghosts/Solution.rb
# @param {Integer[][]} ghosts
# @param {Integer[]} target
# @return {Boolean}
def escape_ghosts(ghosts, target)
    min_distance = 20001

    ghosts.each do |ghost|
        distance = (ghost[0] - target[0]).abs + (ghost[1] - target[1]).abs
        min_distance = distance if distance < min_distance
    end

    return min_distance > target[0].abs + target[1].abs
end

File: Medium/0789-Escape The Ghosts/Solution.rs
impl Solution {
    pub fn escape_ghosts(ghosts: Vec<Vec<i32>>, target: Vec<i32>) -> bool {
        ghosts
            .iter()
            .map(|v| (v[0] - target[0]).abs() + (v[1] - target[1]).abs())
            .min()
            .unwrap_or(20001)
            > (target[0].abs() + target[1].abs())
    }
}
